ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamStreamJoinBuilder.java

```diff
@@ -77,25 +77,26 @@ private StreamStreamJoinBuilder() {
         StreamsUtil.buildOpName(queryContext),
         StreamsUtil.buildOpName(queryContext)
     );
-    final JoinParams joinParams = JoinParamsFactory.create(leftSchema, rightSchema);
+
+    final KsqlValueJoiner joiner = new KsqlValueJoiner(leftSchema, rightSchema);
     final JoinWindows joinWindows = JoinWindows.of(join.getBefore()).after(join.getAfter());
     final KStream<K, GenericRow> result;
     switch (join.getJoinType()) {
       case LEFT:
         result = left.getStream().leftJoin(
-            right.getStream(), joinParams.getJoiner(), joinWindows, joined);
+            right.getStream(), joiner, joinWindows, joined);
         break;
       case OUTER:
         result = left.getStream().outerJoin(
-            right.getStream(), joinParams.getJoiner(), joinWindows, joined);
+            right.getStream(), joiner, joinWindows, joined);
         break;
       case INNER:
         result = left.getStream().join(
-            right.getStream(), joinParams.getJoiner(), joinWindows, joined);
+            right.getStream(), joiner, joinWindows, joined);
         break;
       default:
         throw new IllegalStateException("invalid join type");
     }
-    return left.withStream(result, joinParams.getSchema());
+    return left.withStream(result, join.getSchema());
```

Comment:
Please don't get the schema out of the step classes. I'm working on dropping the schema from them now that we are computing it as we build the stream.

Reply:
Do you mind picking this up when you're doing that? Because at the moment table functions don't work with the joinParams schema here.  

Either that, or I'll tackle as a follow up PR, (but I'd rather crack on with primitive keys).